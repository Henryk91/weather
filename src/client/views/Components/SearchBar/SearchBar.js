import React, { Component } from 'react'
// import './style.css';
export default class SearchBar extends Component {

    constructor(props) {
        super(props);
        this.state = {
            search: false,
        };
        this.getData = this.getData.bind(this);
        this.getCoordinates = this.getCoordinates.bind(this);
        this.useGeoloc = this.useGeoloc.bind(this)

    }
    getData(latitude, longitude) {
        fetch(`/api/weather?coordinates=${latitude},${longitude}`)
            .then(res => res.json())
            .then(data => {
                this.props.set({ weatherData: data })
                this.setState({ search: false })
            });
    }
    getCoordinates() {
        if (this.title.value) {
            var title = this.title.value;
            this.setState({ search: true })
            fetch('https://nominatim.openstreetmap.org/search/' + title + '?format=json&limit=1')
                .then(res => res.json())
                .then(data => {
                    this.getData(data[0].lat, data[0].lon)
                })
        } else {
            alert("Please Enter A Location")
        }
    }
    useGeoloc() {
        this.setState({ search: true })
        fetch('https://ipapi.co/json')
            .then(res => res.json())
            .then(data => {
                this.title.value = data.city;
                this.getData(data.latitude, data.longitude)
            })
    }
    render() {
        return (
            <header>
                <button id="geoButton" className="searchButton blueHover" onClick={this.useGeoloc} title="Geo Locate">
                    <i className="fas fa-map-marker-alt"></i>
                </button>
                <input id="locationBox" type="text" ref={(c) => this.title = c}
                    required="required"aria-label="Type Location Here" placeholder="Type Location..." ></input>
                <button id="goButton" className="searchButton blueHover" onClick={this.getCoordinates} title="Search">
                    <i className="fas fa-search"></i>
                </button>
                {this.state.search ? <div className="loader"></div> : null}
            </header>
        )
    }
}
