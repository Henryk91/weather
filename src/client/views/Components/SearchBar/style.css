#locationBox{
  font-size: 25px;
}
.searchButton{
  height: 30px;
  width: 30px;
}
i{
  margin: 2px;
}
header{
  margin-top: 10px;
}