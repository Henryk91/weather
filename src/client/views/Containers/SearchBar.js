import React, { Component } from 'react'

export default class SearchBar extends Component {
    
    constructor(props) {
        super(props);
        this.state = {
            locationData: null,
            weatherData: null,
            latitude: null,
            longitude: null,
            locationName: "",
            search: false,
        };
        this.getData = this.getData.bind(this);
        this.getCoordinates = this.getCoordinates.bind(this);
        this.useGeoloc = this.useGeoloc.bind(this)

    }
    getData(latitude, longitude) {

        fetch(`/api/weather?coordinates=${latitude},${longitude}`)
            .then(res => res.json())
            .then(data => {
                this.props.set({ weatherData: data })
                this.setState({search: false})
            });
        console.log("Data Recieved")
    }

    getCoordinates() {
        this.setState({search: true})
        var title = "Glasgow";

        if (this.title.value) {
            title = this.title.value;
        } else {
            this.title.value = title;
        }

        fetch('/api/geocode/?query=' + title)
            .then(res => res.json())
            .then(data => {
                this.setState({ locationData: data }),
                this.getData(data.latitude, data.longitude)
            })
    }

    useGeoloc(){
        this.setState({search: true})
        fetch('http://ip-api.com/json')
            .then(res => res.json())
            .then(data => {
                this.setState({ locationData: data }),
                this.title.value = data.city;
                this.getData(data.lat, data.lon)
            })
            
        
    }

    render() {
        
        return (
            <div>
                <br />
                <button className="blueHover" onClick={this.useGeoloc}> Geo </button>
                <input id="locationBox" type="text" ref={(c) => this.title = c} placeholder="Type Location Here..." ></input>
                <button className="blueHover" onClick={this.getCoordinates}> GO </button>
                {this.state.search ? <div className="loader"></div> : null }
            </div>
        )
    }
}
