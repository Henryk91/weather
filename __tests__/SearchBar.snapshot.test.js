
import SearchBar from '../src/client/views/Components/SearchBar/SearchBar';
import renderer from 'react-test-renderer';
import React from 'react';


test('Searchbar component renders the header correctly', () => {
  const rendered = renderer.create(
    <SearchBar />
  );
  expect(rendered.toJSON()).toMatchSnapshot();
});