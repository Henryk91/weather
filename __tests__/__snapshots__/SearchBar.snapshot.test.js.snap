// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchbar component renders the header correctly 1`] = `
<header>
  <button
    className="searchButton blueHover"
    id="geoButton"
    onClick={[Function]}
    title="Geo Locate"
  >
    <i
      className="fas fa-map-marker-alt"
    />
  </button>
  <input
    aria-label="Type Location Here"
    id="locationBox"
    placeholder="Type Location..."
    required="required"
    type="text"
  />
  <button
    className="searchButton blueHover"
    id="goButton"
    onClick={[Function]}
    title="Search"
  >
    <i
      className="fas fa-search"
    />
  </button>
</header>
`;
